agents/template/src/main/scala/org/broadinstitute/transporter/kafka/TransferState.scala

```diff
@@ -0,0 +1,46 @@
+package org.broadinstitute.transporter.kafka
+
+import io.circe.{Encoder, Json}
+import io.circe.syntax._
+import org.broadinstitute.transporter.kafka.TransferStreamBuilder.UnhandledErrorInfo
+import org.broadinstitute.transporter.transfer.TransferResult
+
+/** The possible states of a transfer request as seen by an agent. */
+sealed abstract class TransferState[+I, +P, +O] extends Product with Serializable {
```

Comment:
OR we should have `step` return this too, for consistency. I think we'll want to get there eventually (for the retriable failure case). WDYT?

Reply:
I think it makes sense to have `step` return this too, but maybe that should be another ticket/PR so that all the changes that need to be made for that are done together. I can add a ticket for that.